system-parachains/collectives/collectives-polkadot/src/secretary/mod.rs

```diff
@@ -0,0 +1,207 @@
+	>,
+>;
+
+impl pallet_secretary_origins::Config for Runtime {}
+
+pub struct NoopGetMaxVoters;
+
+impl GetMaxVoters for NoopGetMaxVoters {
+	type Class = u16;
+
+	fn get_max_voters(_c: Self::Class) -> MemberIndex {
+		0 // No-op implementation always returns 0
+	}
+}
+
+pub struct SecretaryPolling<T: pallet_ranked_collective::Config<I>, I: 'static, M: GetMaxVoters>(
+	PhantomData<(T, I, M)>,
+);
+
+impl<T: pallet_ranked_collective::Config<I>, I: 'static, M: GetMaxVoters> Polling<TallyOf<T, I>>
+	for SecretaryPolling<T, I, M>
+{
+	type Index = MemberIndex;
+	type Votes = Votes;
+	type Class = u16;
+	type Moment = BlockNumberFor<T>;
+
+	fn classes() -> Vec<Self::Class> {
+		vec![]
+	}
+
+	fn as_ongoing(_index: Self::Index) -> Option<(TallyOf<T, I>, Self::Class)> {
+		None
+	}
+
+	fn access_poll<R>(
+		_index: Self::Index,
+		f: impl FnOnce(PollStatus<&mut TallyOf<T, I>, Self::Moment, Self::Class>) -> R,
+	) -> R {
+		f(PollStatus::None)
+	}
+
+	fn try_access_poll<R>(
+		_index: Self::Index,
+		f: impl FnOnce(
+			PollStatus<&mut TallyOf<T, I>, Self::Moment, Self::Class>,
+		) -> Result<R, DispatchError>,
+	) -> Result<R, DispatchError> {
+		f(PollStatus::None)
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn create_ongoing(_class: Self::Class) -> Result<Self::Index, ()> {
+		Err(())
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn end_ongoing(_index: Self::Index, _approved: bool) -> Result<(), ()> {
+		Err(())
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn max_ongoing() -> (Self::Class, u32) {
+		(0, 0)
+	}
+}
+
+pub type SecretaryCollectiveInstance = pallet_ranked_collective::Instance3;
+
+impl pallet_ranked_collective::Config<SecretaryCollectiveInstance> for Runtime {
+	type WeightInfo = (); // TODO weights::pallet_ranked_collective_secretary_collective::WeightInfo<Runtime>;
+	type RuntimeEvent = RuntimeEvent;
+	type AddOrigin = ApproveOrigin;
+	type RemoveOrigin = ApproveOrigin;
+	type PromoteOrigin = ApproveOrigin;
+	type DemoteOrigin = ApproveOrigin;
+	type ExchangeOrigin = ApproveOrigin;
+	type Polls = SecretaryPolling<Runtime, SecretaryCollectiveInstance, NoopGetMaxVoters>;
+	type MinRankOfClass = Identity;
+	type MemberSwappedHandler = crate::SecretarySalary;
+	type VoteWeight = pallet_ranked_collective::Geometric;
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkSetup = crate::SecretarySalary;
+}
+
+pub type SecretarySalaryInstance = pallet_salary::Instance3;
+
+parameter_types! {
+	// The interior location on AssetHub for the paying account. This is the Secretary Salary
+	// pallet instance. This sovereign account will need funding.
+	pub SecretarySalaryInteriorLocation: InteriorLocation = PalletInstance(<crate::SecretarySalary as PalletInfoAccess>::index() as u8).into();
+}
+
+const USDT_UNITS: u128 = 1_000_000;
+
+/// [`PayOverXcm`] setup to pay the Secretary salary on the AssetHub in USDT.
+pub type SecretarySalaryPaymaster = PayOverXcm<
+	SecretarySalaryInteriorLocation,
```

Comment:
Maybe better separate account for a separate accounting. But can be funded from the fellowship sub treasury. No strong opinion. Let's hear Joe opinion.

Reply:
Can we control the account using a fellowship proposal?

Reply:
Are the secretaries only for the Fellowship? I thought the program was meant to include secretaries for other collectives as well. If it's just for the Fellowship, then yeah we should just pay them from the Fellowship salary account. But if it's meant to serve others, then I think they can have their own sub-treasury and make proposals for funding to the collectives that they provide secretary services for.

Reply:
I think the current idea is to enable the secretary collective for the technical fellowship for now until more ideas and participants are clearer, rather than having a tresury implementation for just one member. The salary payout account can be changed to the technical fellowship interior account IMO.

Reply:
It seems wasteful (using pallet indices and adding code to maintain) to have an entire new collective just to support one other collective. If secretaries are tied to a collective then we should just change `pallet-ranked-collective` and make a special rank (or just make a rank 10 in the Fellowship and configure the origins so that 10 is "special" (e.g. can open referenda but not vote)).

Adding this as a new collective only seems worth it if it is a general service to all collectives.

Reply:
Yes, it will eventually become a general service to all collectives, adding functionalities for the current collective(Technical fellowship) is currently what is being considered. Other functionalities will come as more collectives are added and I look forward to working on that as well.

Keeping it light weight gives room to more ideas

Reply:
Then to me it sounds like using the Fellowship Salary account is a big mistake because we'll eventually need to add a Secretary one and manage migrating funds to it.

Using the Fellowship salary account doesn't sound like keeping it light, it sounds like creating a heavy migration that requires a new PR and runtime upgrade to make the Secretary Collective usable for other collectives.

Reply:
Can the Secretary collective make proposals to other collectives to fund their main salary account that would exist on AssetHub(USDT)?

Reply:
I would hope so. Either way, if they can't convince someone to open it, it probably wouldn't work out anyway.

Reply:
Final note on this is to change `pallet-ranked-collective` to include a special rank for `Secretary`?

Reply:
@Doordashcon what do you mean?

Reply:
@joepetrowski higlighted that using the Fellowship Salary account would be a wrong direction because of needing migration later on.

Reply:
It's not the "wrong" direction, it's a choice. I see two solutions:

1. Special rank of `Secretary` within Ranked Collective (it doesn't need to be called that in the pallet, could even be a rank 10 for example and the origins are configured to treat that as special);
2. Separate Secretary collective with its own sub-treasury.

In the case of (1), it would naturally just use the specific collective's sub-treasury. In the case of (2), each collective that the Secretary collective serves would have to chip in funds to pay for the Secretary collective's services.

The only thing that is wrong IMO is to make one Secretary collective for every other collective (effectively halving the total number of collectives that the chain can support).

Reply:
Thank you for the clarification, @muharem suggested we don't need the Tresury implementation, the Secretary Salary account can be funded using the Technical fellowship sub-treasury. 

When more members join who perform operations for other collectives, they can also fund the Salary account using their respective sub-tresuries, a blocker to this might be the refusal of those collectives to comply.

But a need for another Secretary member would have to arise before considering adding another member to the collective.

We won't have a Secretary collective for every other collective, just the one and add more member through the demand for another that serves a seperate collective.

I hope I have been able to understand the suggestions clearly so far 🙏.

Reply:
Yes, we do not need a sub treasury to pay salaries or fund the Secretary salary account. The whole Secretary can be setup as ranked collective pallet and salary pallet and can serve later secretaries for all collectives. We can even have a different salary accounts for different secretary ranks if needed later. They can be funded by other collective's sub treasuries or the Polkadot Treasury. We do not need referenda and origins because in all cases right now we just need to check if the account belongs to a certain rank of the collective, no plurality voice needed.